File: common/src/main/java/io/muun/common/crypto/MuunAddressGroup.java
package io.muun.common.crypto;

import io.muun.common.crypto.hd.MuunAddress;

public class MuunAddressGroup {

    public final MuunAddress legacy;
    public final MuunAddress segwit;

    /**
     * Constructor.
     */
    public MuunAddressGroup(MuunAddress legacy, MuunAddress segwit) {
        this.legacy = legacy;
        this.segwit = segwit;
    }
}

File: android/apolloui/src/main/java/io/muun/apollo/domain/model/MuunAddressGroup.kt
package io.muun.apollo.domain.model

import io.muun.common.crypto.hd.MuunAddress

class MuunAddressGroup(
    val legacy: MuunAddress,
    val segwit: MuunAddress,
    val taproot: MuunAddress,
) {

    fun toAddressGroup() =
        AddressGroup(legacy.address, segwit.address, taproot.address)
}

class AddressGroup(
    val legacy: String,
    val segwit: String,
    val taproot: String,
) {

    fun getAddress(addressType: AddressType) =
        when (addressType) {
            AddressType.LEGACY -> legacy
            AddressType.SEGWIT -> segwit
            AddressType.TAPROOT -> taproot
        }
}